sdk/core/Azure.Core/src/Shared/TaskExtensions.cs

```diff
@@ -87,5 +91,39 @@ private static void VerifyTaskCompleted(bool isCompleted)
                 throw new InvalidOperationException("Task is not completed");
             }
         }
+
+        public readonly struct Enumerable<T> : IEnumerable<T>
```

Comment:
As public types, should these be fully commented?

Reply:
They aren't public cause `TaskExtensions` is internal

Reply:
Fair enough... I overlooked that.  My bad.